pkg/dynamiclistener/server.go

```diff
@@ -404,6 +424,21 @@ func (s *Server) getCertificate(hello *tls.ClientHelloInfo) (*tls.Certificate, e
 	return tlsCert, nil
 }
 
+func (s *Server) getRotateSettingInDays() (int64, error) {
+	setting, err := s.settingsLister.Get("", settings.RotateCertsIfExpiringInDays.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return 0, err
+	}
+	if setting == nil || setting.Value == "" {
+		return 7, nil
+	}
+	certRotateSetting, err := strconv.ParseInt(setting.Value, 10, 64)
+	if err != nil {
+		return 0, err
```

Comment:
same here. Any error should result in default

Reply:
oh but we won't use that value anywhere, even from getCertificate we will return this error and not use this in evaluating how much time is left, plus it's not updating the setting in etcd either.